src/implementations/transaction/transaction.ts

```diff
@@ -0,0 +1,45 @@
+import signale from "signale";
+import { TransactionCancelCommand } from "../../commands/transaction/transaction-command.type";
+import { getWallet } from "../utils/wallet";
+import { BigNumber } from "ethers";
+
+/*
+  The trick to “cancel” your pending transaction is by replacing the transaction with 
+  another 0 ETH transaction with a higher gas fee sending to yourself with the same nonce as the 
+  pending transaction. By having increasing the gas fee, it allow the miner to prioritize transactions 
+  that pay a higher gas fee.
+  (https://info.etherscan.com/how-to-cancel-ethereum-pending-transactions/)
+*/
+export const cancelTransaction = async ({
+  network,
+  key,
+  keyFile,
+  encryptedWalletPath,
+  nonce,
+  gasPrice,
+  transactionHash,
+}: TransactionCancelCommand): Promise<void> => {
+  try {
+    const wallet = await getWallet({ key, keyFile, network, encryptedWalletPath });
+
+    if (transactionHash) {
+      const currentTransaction = await wallet.provider.getTransaction(transactionHash);
+      nonce = currentTransaction.nonce.toString();
```

Comment:
Never (almost) update function argument. Technically there is no harm dinner here, however it's a bad habit and it's better to not do it at all

Issues can happen if you mutate array or object, where changes will apply in the callet side

Let's introduce local variable, we can even rename the variable from the function to keep using nonce and gasPrice

Reply:
Done.